code/game/machinery/frame.dm

```diff
@@ -231,7 +231,6 @@
 	icon_state = frame_type.get_icon_state(state)
 
 /obj/structure/frame/proc/check_components(mob/user as mob)
-	components = list()
```

Comment:
This line is what empties the list of components when the frame is emptied of components.  Without it, emptying the frame of components teleports every component that has ever been in the frame to the location of the frame, even if you've taken those parts out and used them to build a different machine.